### DISCUSSION TITLE
Can't install Alba Braunschweig?

### DISCUSSION THREAD

**Discussion by barthel-eu** [mampfes/hacs_waste_collection_schedule#2331] [Q&A] [answered]
I installed Waste Collection Schedule in HACS. When I try to add the integration in HA, it asks me the service (Germany - Alba Braunschweig), then gives me a long questionaire that is nowhere explained where I looked (see below)?

https://github.com/mampfes/hacs_waste_collection_schedule/blob/2.0.1/doc/ics/alba_bs_de.md gives me the ability to adjust the URL for my address. But it doesn't tell me where to add that code (and why, given the claim that it's a UI-setup?), entering the URL into the form provided in the addition of the integration to my HA, it tells me Alba returned an empty result.
I tried both with (mf-trash-thisyear%5D={%Y} and mf-trash-thisyear%5D=2024 and tried 2024 also in the Year Field (see below)

The form provided in the process to add the integration asks for:

Calendar Title: [prefilled: ICS]
URL: [I entered the modified URL from above link]
File Name: [?]
Offset: [???]
Parameters: [?]
Year Field: [I tried 2024 to no change]
Method: [prefilled: GET]
Regex: [?]
Titel Template: [prefilled: {{date.summary}}
Split At: [?]
Version: [prefilled: 2]
Verify SSL [checked]
Headers [?]

Submit returns the error: _The source returned an empty response. Please check the provided arguments and try again._

I would very much appreciate, if there is a step by step howto, and/or some explanation, what information is required in those fields. As it was reported in earlier (closed) posts, https://alba-bs.de/service/abfuhrtermine.html does not return a useful GET-link but immediately tries to download the ICS file ALBA-Entsorgungskalender-2024.ics
IMHO https://github.com/mampfes/hacs_waste_collection_schedule/blob/2.0.1/doc/ics/alba_bs_de.md should provide information on what information must be entered within the process of the UI-driven setup?

As it is, I can't succeed with the very basic setup of Waste Collection Schedule but get stalled in the above form. I also learned, that it is very important to have frequent updates, as there have been ad hoc changes to our schedule, which is why I looked if there is a possibility to add them additionally (to the use of their app) into HA.
Also links to where I'd find that missing information and explanation would be very much welcome!

Thanks a million!!

**Comment by 5ila5** [comment, marked as answer]
[alba_bs_de](https://github.com/mampfes/hacs_waste_collection_schedule/blob/2.0.1/doc/ics/alba_bs_de.md) is not a real source but just an alias + some documentation for the ICS source like stated in the first section of the md file.
> ALBA Braunschweig is supported by the generic [ICS source](https://github.com/mampfes/hacs_waste_collection_schedule/blob/2.0.1/doc/source/ics.md). For all available configuration options, please refer to the source description.

So you can look in the ICS source for documentation on all the available input fields. But As you can see, alba_bs_de only needs the URL, so you can leave the other fields on their default value (may also be empty)

You can get the Link by clicking on the `ICS-Kalender-Datei` button and right-click -> copy ULR the URL of the ok button of the popup

You should then like stated in the documentation replace your year with `{%Y}`

If you add the ICS file with the URL it is fetched daily.

But if you prefer the app API: it looks like the app might be supported by [app_abfallplus_de source](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/source/app_abfallplus_de.md)

**Comment by antarxis** [comment]
Unfortunately it is not well documented. Neither for the country/city method nor for the custom one. 

As listed in your post - there are many mandatory fields, but info on what they should contain is close to none. And not having them filled is a blocker. I have been also struggling for hours now to get it working for Braunschweig as well as another city, no success. 

Following the method from 5ila5 didn't help either: I can get the link, but putting it into URL field simply does not work. I have already tried many combinations. Different combinations, different error messages with the same result: no result.

The idea of the add-on is very good, but in this form it is unusable to me. I could imagine having the add-on config file (not the main configuration.yaml to jeopardize the whole running HA) as for many other addons, which can be edited like with:

waste_collection_schedule:
  sources:
    - name: ics
      args:
        url: https://alba-bs.de/service/abfuhrtermine/ajax-kalender.html?tx_mfabfallkalender_mfabfallkalender%5Baction%5D=datelist&tx_mfabfallkalender_mfabfallkalender%5Bcontroller%5D=Abfallkalender&tx_mfabfallkalender_mfabfallkalender%5Bmf-trash-thisyear%5D={%Y}-trash-hausnr%5D=13&tx_mfabfallkalender_mfabfallkalender%5Bmf-trash-hausnrzusatz%5D=&tx_mfabfallkalender_mfabfallkalender%5Bmf-trash-strasse%5D=...
        
Maybe this could just work better. I don't know. But there is no such option being each time stuck with the dialog window, which cannot be filled in in a satisfactory way.

**Reply by 5ila5** [reply]
I know it's not ideal to show all ICS source options for sources that only require the URL, but the configuration menu should display some documentation on how to retrieve the URL and to use this as the URL parameter. I want to add the options to show custom configuration dialogs it would probably be a good to limit the input options for these ICS sources, but I currently do not have the time to work on that.

As for the configuration, I don't quite know hot this should work with a dedicated configuration source and what should be the improvement. HA needs to know about this configuration to load it, so it either needs to be configured in the configuration.yaml or in a configuration file loaded by the configuration.yaml.

The example you printed should behave exactly like, just inserting the URL in the URL field of the ICS source in the GUI.

We currently have some dependency issues if you install another integration requireing icalendar >= 6.0.0 like the ICS Calendar integration.

If you have issues with that please take a look here: https://github.com/mampfes/hacs_waste_collection_schedule/issues/2893#issuecomment-2433200004

**Comment by antarxis** [comment]
Many thanks for your reply. I know the problem of time very well :) 
I attach the error msg I get, maybe it sounds already familiar to you? If not, I will keep trying, I"d like to get the functionality working. It is still possible it is sth on ALBA's end..
Thank you for your effort and support.
![HA_waste](https://github.com/user-attachments/assets/0ea3c6ec-6a31-4fe7-9815-576ca1a1112c)

**Reply by 5ila5** [reply]
The error message indicates your URL leads to an HTML page instead of to an ICS file

You should be able to put the url in a browser which should then prompt you to download the ICS file immediately and not lead to a classic web page

**Comment by antarxis** [comment]
![Screenshot 2024-12-01 at 00-59-36 Ajax Kalender](https://github.com/user-attachments/assets/cbb8908b-88f0-482c-a6d7-35af4b4ce1eb)
It helped. many thanks!

Maybe it can be of some help to others, so I will describe the reason briefly. The case was, that ALBA webpage returned only https://alba-bs.de/# as the link to .ics file. I managed to capture the link in the screenshot, but that one was also not the right one, because ALBA set an intermediate website with some info (screenshot above), and only after clicking "OK" there, the actual ics download followed. So with debug view in browser I finally managed to capture the correct link and now could get through the add on config! 

Many thanks @5ila5!

**Reply by 5ila5** [reply]
I improved the documentation to better explain this step: #3126
